### ISSUE TITLE
LX200 ACK is not handled correctly in ESP firmware

### ISSUE THREAD

**Issue by bgilsrud** [hjd1964/OnStepESPServer#1]
The LX200 ACK command (0x6) appears to be buffered in the ESP firmware until any other LX200 command is sent. It looks like the ESP firmware sends the command to the OnStep controller once it sees the '#' character...which isn't provided with the ACK command.

**Comment by bgilsrud** [comment]
I suspect that the following would fix the issue:
diff --git a/OnStepESPServer.ino b/OnStepESPServer.ino
index 9e7fa6d..a05adf6 100644
--- a/OnStepESPServer.ino
+++ b/OnStepESPServer.ino
@@ -367,7 +367,7 @@ void loop(void){
     writeBuffer[writeBufferPos]=b; writeBufferPos++; if (writeBufferPos>39) writeBufferPos=39; writeBuffer[writeBufferPos]=0;
 
     // send cmd and pickup the response
-    if (b=='#') {
+    if (b=='#' || ((strlen(writeBuffer) == 1) && (b == (char) 6))) {
       char readBuffer[40]="";
       readLX200Bytes(writeBuffer,readBuffer,CmdTimeout); writeBuffer[0]=0; writeBufferPos=0;


I can test this in the next couple of days.

**Comment by hjd1964** [comment]
Thanks.  I'll make that change to both the OnStepESPServer and OnStepEthernetServer once you've tested.

**Comment by bgilsrud** [comment]
I tested this last night and the ACK command is now working as expected. I think this change depends on the timeout during the response read in readLX200Bytes since the ACK response also doesn't include a trailing '#' character...but it does work.

That said, I did experience some strangeness with my mount while using the ESP server interface. Using the android app, GOTOs would start and then only go a couple of degrees before stopping no matter the target. This was after using the LX200 OnStep INDI driver via the WIFI interface (of which I'm clearly the first person do so) so it's possible that it sent something strange to OnStep which confused it. I didn't debug the problem because I wanted to do some actual imaging.

The syntax for the ACK command was an unfortunate choice by Meade...

**Comment by hjd1964** [comment]
**The syntax for the ACK command was an unfortunate choice by Meade**
Indeed.  I actually made the decision to not bother covering ACK in the WiFi/Ethernet servers at the time I wrote that, not something most software uses anyway.  I was more worried about making the LX200 pass-through smart/efficient enough.

Speaking of smart/efficient LX200 pass-through, I see you noticed the timeout in readLX200Bytes() is forced, given the infrequent use of ACK I doubt anyone would ever notice but I added code to handle that too.  I have a test OnStepEthernetServer setup at the moment and it continues to work fine with this improvement though I don't really use the ACK command.

Both OnStepESPServer and OnStepEthernetServer have been updated with this.
